includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -716,4 +814,17 @@ protected function maybe_process_subscription_early_renewal_failure( $order, $in
 			exit;
 		}
 	}
+
+	/**
+	 * Returns true if a subscription payment must be authorized by the customer off session.
+	 *
+	 * This is only valid when using mandates for Indian 3DS regulations.
+	 *
+	 * @param StdClass $payment_intent the Payment Intent to be evaluated.
+	 * @return bool true if payment intent must be authorized off session, false otherwise.
+	 */
+	protected function must_authorize_off_session( $payment_intent ) {
+		return isset( $payment_intent->next_action->type )
+			&& 'card_await_notification' === $payment_intent->next_action->type;
```

Comment:
According to the [documentation](https://stripe.com/docs/india-recurring-payments#trigger-predebit-notification) this should be changed:

```suggestion
		return isset( $payment_intent->processing->card->customer_notification->approval_requested )
			&& filter_var( $payment_intent->processing->card->customer_notification->approval_requested, FILTER_VALIDATE_BOOL );
```

However, the API responses I'm getting from the API don't match the documentation so I've changed it to work even if it doesn't seem to be correct. I've reached out to Stripe in p1678862901758999-slack-C9976E5MJ to ask for clarification.

Reply:
Previous suggestion was wrong, this is what's required:

```suggestion
		return ! empty( $payment_intent->status )
			&& 'processing' === $payment_intent->status
			&& ! empty( $payment_intent->processing->card->customer_notification->completed_at );
```

Reply:
Committed the suggestion since it's more likely people will have this version of the API than what I'm seeing personally.